### PR TITLE
Create adding_an_app

### PR DESCRIPTION
**Pull request by caljess599**
Describes how to install an app to your Harvest project. Gives example of the import_export tool.

### CODE REVIEW

**Comment by caljess599** [comment]
Hope this is somewhat useful. Perhaps it could be linked to from the 'Using Your Data' page?

**Comment by bruth** [comment]
@caljess599 Thank you for putting this together. I am looking forward to putting promoting this in the new bi-monthly newsletter that goes how on Monday. I am not sure how familiar you are with Github, but I am going to leave comments as I review on suggested changes on style or clarification on the content.

File: _posts/adding_an_app [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+This is a handy tool that allows the user to easily find and manage virtual environments. [Installation instructions](http://virtualenvwrapper.readthedocs.org/en/latest/install.html)
+
+Harvest works nicely with this tool: once you have installed and configured this tool, you can `init` a Harvest project with the `--venv-wrap` option and it will place the virtual environment where you've specified in your virtualenvwarpper configuration. That makes it really easy to find and control the environment.
+
+##Where and how to install a Python package
+If your package is on the Python Package Index (PyPI), then installation can be done through `pip`, the same command-line tool that you used to install Harvest. [Installation instructions](https://django-import-export.readthedocs.org/en/latest/installation.html)
+
+Simply navigate to the directory where your virtual environments are located and activate (if the first time) or bring up (the virtualenvwrapper command is `workon projectname`. When the virtual environment is up, you can now install packages and they will go into that virtual environment.
+
+## Adding your app to the list of install apps
```
Comment (bruth) [inline _posts/adding_an_app]: Refer to "install apps" as `INSTALLED_APPS` (use backticks in Markdown to display as code) so it's clear it referring to a setting.

File: _posts/adding_an_app [hunk 2]
```diff
@@ -0,0 +1,48 @@
+#Adding an app to your Harvest instance
```
Comment (bruth) [inline _posts/adding_an_app]: For readability, please add a space after the hash signs in headers, e.g. `# Adding an app...`

File: _posts/adding_an_app [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+##Where and how to install a Python package
+If your package is on the Python Package Index (PyPI), then installation can be done through `pip`, the same command-line tool that you used to install Harvest. [Installation instructions](https://django-import-export.readthedocs.org/en/latest/installation.html)
+
+Simply navigate to the directory where your virtual environments are located and activate (if the first time) or bring up (the virtualenvwrapper command is `workon projectname`. When the virtual environment is up, you can now install packages and they will go into that virtual environment.
+
+## Adding your app to the list of install apps
+Now, in your Harvest project's `global_settings.py` file, you'll need to add the name of your app to the list of installed apps. Then refer to the documentation of your particular package as to what else you need to create/modify in order to access the functionality of your specific package.
+
+##A few specifics about the import_export package
```
Comment (bruth) [inline _posts/adding_an_app]: I would recommend adding a section before this introducing the package and briefly describing your need for it.

File: _posts/adding_an_app [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+As this is an admin tool, you'll need to create (if you haven't already done so) an `admin.py` file among your project files.
+
+The ResourceClass and AdminClass that you specify in this file allows the import_export tool how to work with your model.
+
+```
```
Comment (bruth) [inline _posts/adding_an_app]: Append `python` to the triple backticks to get syntax highlighting.

**Comment by caljess599** [comment]
Yes, please do help me get the Github format correct--I was struggling a
bit with the markup... Happy to learn/amend/improve!

**Comment by naegelyd** [comment]
@caljess599 Thanks for updating this pull request. Can you please squash this into a single commit?

**Comment by caljess599** [comment]
I would gladly squash them but sadly I don't know how. Very new to editing
files on git. I did all of this through the web interface and I cannot
figure out how to squash the two commits into one. (I do see them, though.)
I tried to google for help but the only info I find is doing it through the
command line, which I believe only applies if I cloned the repo locally,
which I didn't do.

**Comment by bruth** [comment]
Not a problem. We will handle the few remaining changes needed before merging. If you have interest in learning how to use git on the command line I can point you to some resources.

**Comment by caljess599** [comment]
Thanks, Byron. I am beginning to use git on the command line. I will
certainly look at the squash command in this context (I take it you can't
do things like that on the web UI...)
Jessica

**Comment by bruth** [comment]
GitHub does not provide any controls for _history altering_ operations such as a squash or rebase since they can be misused pretty easily.

**Comment by bruth** [comment]
@caljess599 I merged your commits in the main repo with a few additional tweaks. You can see a preview here: http://dev.harvest.io/articles/2014/10/30/adding-an-app/

We are going to review and edit it further, but thanks again for putting this together. Of course if you want to add to the article or make changes yourself, you can create another branch.

**Pull request closed without merging**